src/odemis/gui/cont/streams.py

```diff
@@ -3335,12 +3337,19 @@ def _on_combobox(self, _):
     def _on_text(self, evt):
         txt = self.panel.txt_ctrl.GetValue()
         current_name = self.model.name.value
-        all_roas = [roa.name.value for project in self._tab_data.projects.value for roa in project.roas.value]
+
+        # Process input, make sure name is unique in project and complies with whitelisted characters of
+        # technolution driver
+        roa_project = [p for p in self._tab_data.projects.value if self.model in p.roas.value][0]
+        all_project_roas = [roa.name.value for roa in roa_project.roas.value]
+        txt = txt.replace(" ", "-")  # replace spaces with dash
+        txt = re.sub(r'[^A-Za-z0-9_()-]+', '', txt)  # only allow characters from whitelist
```

Comment:
Possibly a more elegant way would be to use the notion of "Validator" on the wx.TextCtrl. Essentially, this would prevent the user from typing text which is not acceptable. 
It would be a matter of creating a Validator, and then calling textctrl.SetValidator(...).  You can have a look at https://www.wxpython.org/Phoenix/docs/html/validator_overview.html , and the class odemis.gui.comp.text._NumberValidator .

Reply:
done